pkg/reconciler/pipelinerun/resources/apply.go

```diff
@@ -251,38 +251,56 @@ func ApplyReplacements(p *v1beta1.PipelineSpec, replacements map[string]string,
 		if p.Finally[i].TaskRef != nil && p.Finally[i].TaskRef.Params != nil {
 			p.Finally[i].TaskRef.Params = replaceParamValues(p.Finally[i].TaskRef.Params, replacements, arrayReplacements, objectReplacements)
 		}
-		p.Finally[i], replacements, arrayReplacements, objectReplacements = propagateParams(p.Finally[i], replacements, arrayReplacements, objectReplacements)
+		p.Finally[i] = propagateParams(p.Finally[i], replacements, arrayReplacements, objectReplacements)
```

Comment:
can you please also add a test case covering finally tasks?

Reply:
Ok, I added a test case to cover finally tasks